test/parse.js

```diff
@@ -34,17 +34,31 @@ test('parse()', function (t) {
 
     t.test('arrayFormat: brackets allows only explicit arrays', function (st) {
         st.deepEqual(qs.parse('a[]=b&a[]=c', { arrayFormat: 'brackets' }), { a: ['b', 'c'] });
+        st.deepEqual(qs.parse('a[0]=b&a[1]=c', { arrayFormat: 'brackets' }), { 'a[0]': 'b', 'a[1]': 'c' });
```

Comment:
i see how this seems correct, because it's currently set to brackets but is parsing indices.

However, the current behavior matches what PHP, Rails, and express all do, so I'm not sure it should change.

Reply:
yeah, i'm not very sure on this one too, though you have to explicitly state you want this behavior so maybe it's ok

alternatively making allowing `arrayFormat` to be an array itself could solve it

```js
t.deepEqual(qs.parse('a[0]=b&a[1]=c', { arrayFormat: ['brackets', 'indices'] }), { 'a': ['b',  'c'] });
t.deepEqual(qs.parse('a[]=b&a[]=c', { arrayFormat: ['brackets', 'indices'] }), { 'a': ['b',  'c'] });
```